server/src/main/java/org/opensearch/index/codec/composite/datacube/startree/fileformats/StarTreeWriter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.index.codec.composite.datacube.startree.fileformats;
+
+import org.apache.lucene.store.IndexOutput;
+import org.opensearch.index.codec.composite.datacube.startree.fileformats.data.StarTreeDataWriter;
+import org.opensearch.index.codec.composite.datacube.startree.fileformats.meta.StarTreeMetaWriter;
+import org.opensearch.index.compositeindex.datacube.startree.StarTreeField;
+import org.opensearch.index.compositeindex.datacube.startree.aggregators.MetricAggregatorInfo;
+import org.opensearch.index.compositeindex.datacube.startree.node.InMemoryTreeNode;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Util class for building star tree
+ *
+ * @opensearch.experimental
+ */
+public class StarTreeWriter {
+
+    private StarTreeWriter() {}
+
```

Comment:
What is the purpose of this wrapper. Would prefer avoiding `static` utilities, they tend to become kitchen sink. In this case I don't see a strong reason to go with it

Reply:
Based on your comment below, we had to maintain the version for the writer, while decoupling the file formats from the codec as well. (eg: BKDWriter)

With this [change](https://github.com/opensearch-project/OpenSearch/pull/14809/files#diff-9220cbcb11aa9d89e366c682bf224543e86e98f09e2f62687304a9901a9e26e1R28), we can ensure that star tree data and meta are written with the latest version.

During reads, we can optimize or change the implementation based on the version (something similar to this with [BKDReader](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/util/bkd/BKDReader.java#L117))